Content.Client/SurveillanceCamera/UI/SurveillanceCameraMonitorWindow.xaml.cs

```diff
@@ -0,0 +1,188 @@
+using Content.Shared.SurveillanceCamera;
+using Robust.Client.AutoGenerated;
+using Robust.Client.Graphics;
+using Robust.Client.ResourceManagement;
+using Robust.Client.UserInterface.Controls;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Prototypes;
+
+namespace Content.Client.SurveillanceCamera.UI;
+
+[GenerateTypedNameReferences]
+public sealed partial class SurveillanceCameraMonitorWindow : DefaultWindow
+{
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+
+    public event Action<string>? CameraSelected;
+    public event Action<string>? SubnetOpened;
+    public event Action? CameraRefresh;
+    public event Action? SubnetRefresh;
+    public event Action? CameraSwitchTimer;
+    public event Action? CameraDisconnect;
+
+    private string _currentAddress = string.Empty;
+    private readonly FixedEye _defaultEye = new();
+    private readonly Dictionary<string, int> _subnetMap = new();
+
+    private string? SelectedSubnet
+    {
+        get
+        {
+            if (SubnetSelector.ItemCount == 0
+                || SubnetSelector.SelectedMetadata == null)
+            {
+                return null;
+            }
+
+            return (string) SubnetSelector.SelectedMetadata;
+        }
+    }
+
+    public SurveillanceCameraMonitorWindow()
+    {
+        RobustXamlLoader.Load(this);
+        IoCManager.InjectDependencies(this);
+
+        // This could be done better. I don't want to deal with stylesheets at the moment.
+        var texture = IoCManager.Resolve<IResourceCache>().GetTexture("/Textures/Interface/Nano/square_black.png");
+        var shader = IoCManager.Resolve<IPrototypeManager>().Index<ShaderPrototype>("CameraStatic").Instance().Duplicate();
+
+        CameraView.ViewportSize = new Vector2i(500, 500);
+        CameraView.Eye = _defaultEye; // sure
+        CameraViewBackground.Stretch = TextureRect.StretchMode.Scale;
+        CameraViewBackground.Texture = texture;
+        CameraViewBackground.ShaderOverride = shader;
+
+        SubnetList.OnItemSelected += OnSubnetListSelect;
+
+        SubnetSelector.OnItemSelected += args =>
+        {
+            // piss
+            SubnetOpened!((string) args.Button.GetItemMetadata(args.Id)!);
+        };
+        SubnetRefreshButton.OnPressed += _ => SubnetRefresh!();
+        CameraRefreshButton.OnPressed += _ => CameraRefresh!();
+        CameraDisconnectButton.OnPressed += _ => CameraDisconnect!();
+    }
+
+
+    // The UI class should get the eye from the entity, and then
+    // pass it here so that the UI can change its view.
+    public void UpdateState(IEye? eye, HashSet<string> subnets, string activeAddress, string activeSubnet, Dictionary<string, string> cameras)
+    {
+        _currentAddress = activeAddress;
+        SetCameraView(eye);
+
+        if (subnets.Count == 0)
+        {
+            SubnetSelector.AddItem(string.Empty);
```

Comment:
Would a localized string that said "No Subnets" look better?

Reply:
yeah, most likely